crates/datagen/src/lib.rs

```diff
@@ -85,6 +86,61 @@ fn range_as_f64(
     }
 }
 
+fn decimal_max_range(field: &Field) -> AnyResult<(Decimal, Decimal)> {
+    let precision = field
+        .columntype
+        .precision
+        .unwrap_or(FELDERA_MAX_DECIMAL_PRECISION as i64) as u32;
+    if precision > FELDERA_MAX_DECIMAL_PRECISION {
+        return Err(anyhow!(
+            "Invalid precision for field {:?}, max precision is {}",
+            field.name,
+            FELDERA_MAX_DECIMAL_PRECISION
+        ));
+    }
+
+    let scale = field.columntype.scale.unwrap_or(0) as u32;
+    if scale > FELDERA_MAX_DECIMAL_SCALE {
+        return Err(anyhow!(
+            "Invalid scale for field {:?}, max scale is {}",
+            field.name,
+            FELDERA_MAX_DECIMAL_SCALE
+        ));
+    }
+    if scale > precision {
+        return Err(anyhow!(
+            "Invalid scale for field {:?}, scale must be less or equal than precision",
+            field.name
+        ));
+    }
+
+    Ok((decimal_min(precision, scale), decimal_max(precision, scale)))
+}
+
+fn range_as_decimal(
+    name: &SqlIdentifier,
+    range: &Option<(Value, Value)>,
+) -> AnyResult<Option<(Decimal, Decimal)>> {
+    match range {
+        None => Ok(None),
+        Some((a, b)) => {
+            let a = serde_json::from_value::<Decimal>(a.clone()).map_err(|_| {
+                anyhow!(
+                    "Invalid range, min is not a valid decimal for field {:?}",
+                    name
+                )
+            })?;
+            let b = serde_json::from_value::<Decimal>(b.clone()).map_err(|_| {
+                anyhow!(
+                    "Invalid range, max is not a valid decimal for field {:?}",
+                    name
+                )
+            })?;
+            Ok(Some((a, b)))
```

Comment:
does someone else check that the range is valid?
or maybe it doesn't matter?

Reply:
It's intentional in that datagen can also generate out of erroneous / out of range values if the user requests it/wants to e.g., test error behavior (except by default it will respect the range now after this PR)